diff --color -ur a/kcompat-generator.sh b/kcompat-generator.sh
--- a/kcompat-generator.sh
+++ b/kcompat-generator.sh
@@ -147,6 +147,7 @@
 	gen HAVE_NETDEV_XMIT_MORE if fun netdev_xmit_more in "$ndh"
 	gen HAVE_NDO_ETH_IOCTL if fun ndo_eth_ioctl in "$ndh"
 	gen HAVE_NDO_FDB_ADD_VID    if method ndo_fdb_del of net_device_ops matches 'u16 vid' in "$ndh"
+	gen HAVE_NDO_FDB_ADD_NOTIFIED if method ndo_fdb_add of net_device_ops matches 'bool \\*notified' in "$ndh"
 	gen HAVE_NDO_FDB_DEL_EXTACK if method ndo_fdb_del of net_device_ops matches ext_ack in "$ndh"
 	gen HAVE_NDO_GET_DEVLINK_PORT if method ndo_get_devlink_port of net_device_ops in "$ndh"
 	gen HAVE_NDO_UDP_TUNNEL_CALLBACK if method ndo_udp_tunnel_add of net_device_ops in "$ndh"
diff --color -ur a/ngbe_main.c b/ngbe_main.c
--- a/ngbe_main.c
+++ b/ngbe_main.c
@@ -9134,6 +9134,9 @@
 #endif
 #ifdef HAVE_NDO_FDB_ADD_EXTACK
 			     u16 flags,
+#ifdef HAVE_NDO_FDB_ADD_NOTIFIED
+			     bool *notified,
+#endif /* HAVE_NDO_FDB_ADD_NOTIFIED */
 			     struct netlink_ext_ack __always_unused *extack)
 #else
 			     u16 flags)
